pages/embedded-portal.tsx

```diff
@@ -264,16 +254,61 @@ export const getServerSideProps: GetServerSideProps = async (context) => {
     },
   });
 
+  if (!existingSpace) {
+    console.log("No space");
+    return {
+      props: {
+        environmentToken,
+        lastSyncedAt: lastSync
+          ? DateTime.fromJSDate(lastSync.createdAt).toFormat(
+              "MM/dd/yy hh:mm:ss a"
+            )
+          : "",
+        existingSpace,
+        userId: token.sub,
+      },
+    };
+  }
+
+  const spaceData = await getSpace({
+    workflow: WorkflowType.Embedded,
+    spaceId: (existingSpace?.flatfileData as unknown as FlatfileSpaceData).id,
+  });
+
+  console.log("spaceData", spaceData);
+
+  const workbook = await getWorkbook({
+    workflow: WorkflowType.Embedded,
+    workbookId: spaceData?.primaryWorkbookId as string,
```

Comment:
is primaryWorkbookId not guaranteed here?

Reply:
primaryWorkbookId is not typed onto the space. On the config, the space is getting updated with the primaryWorkbookId
![CleanShot 2023-06-13 at 17 37 07](https://github.com/FlatFilers/HCMShow/assets/12041898/dca50667-37bf-4a7f-8984-04328b8acb83)
<img width="577" alt="CleanShot 2023-06-13 at 17 37 18@2x" src="https://github.com/FlatFilers/HCMShow/assets/12041898/14d5b490-4841-4a93-8bf3-8911ef4b703b">